terraform_docs runtime failure  "panic: runtime error: invalid memory address or nil pointer dereference"

<!-- Please note, this template is for bugs report, not feature requests -->
<!-- For more information, see the Contributing Guidelines at -->
<!-- https://github.com/terraform-docs/terraform-docs/tree/master/CONTRIBUTING.md -->

**Describe the bug**

terraform_docs runtime failure  "panic: runtime error: invalid memory address or nil pointer dereference". 
running via precommit hooks. its being invoked from microsoft azure devops pipeline. 

Starting: Run pre-commit-terraform on /home/vsts/work/1/s/terraform/hhck-webapp
==============================================================================
Task         : Command line
Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
Version      : 2.164.2
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
==============================================================================
Generating script.
========================== Starting Command Output ===========================
/bin/bash --noprofile --norc /home/vsts/work/_temp/158cd010-1a06-4dea-a6d6-50c0f7ed0314.sh
pre-commit run --files *
[INFO] Initializing environment for git://github.com/antonbabenko/pre-commit-terraform.
Terraform docs...........................................................Failed
hookid: terraform_docs

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x7de744]

goroutine 1 [running]:
github.com/terraform-docs/terraform-docs/internal/cli.(*cfgreader).exist(0xc0000b41c0, 0xc0000b41c0, 0x2, 0xc00001cf40)
	/home/runner/work/terraform-docs/terraform-docs/internal/cli/reader.go:22 +0xe4
github.com/terraform-docs/terraform-docs/internal/cli.PreRunEFunc.func1(0xc00008fb80, 0xc000112240, 0x1, 0x1, 0x0, 0x0)
	/home/runner/work/terraform-docs/terraform-docs/internal/cli/run.go:46 +0x1b9
github.com/spf13/cobra.(*Command).execute(0xc00008fb80, 0xc000112220, 0x1, 0x1, 0xc00008fb80, 0xc000112220)

**To Reproduce**

Steps to reproduce the behavior:

1. download curl -Lo ./terraform-docs https://github.com/terraform-docs/terraform-docs/releases/download/v0.10.0-rc.1/terraform-docs-v0.10.0-rc.1-$(uname | tr '[:upper:]' '[:lower:]')-amd64
2. configure terraform_docs from precommit
3. execute precommit.


**Expected behavior**

terraform_docs should run successfully and return passed status without throwing runtime errors.
**Version information:**

- terraform-docs version (use `terraform-docs --version`): **v0.10.0-rc.1**
- Go version (if you manually built. use `go version`): **1.14.4**
- OS (e.g. Windows, MacOS): **Ubuntu** 


Thanks in advance

Cheers
Omkar

This happens to me with v0.10.0.  Here are my repro steps (from a temp dir):

1. `git clone https://github.com/terraform-aws-modules/terraform-aws-sns .`
2. `echo >> variables.tf`
3. `git add variables.tf`
4. `pre-commit`

Here's what the pre-commit hook looks like when I run it through with `bash -x`
<details>
<summary>Logs</summary>
```bash
$ bash -x ~/.cache/pre-commit/repo6u_wz1ng/terraform_docs.sh . --args="table"
+ set -e
+ [[ /Users/chris.przybycien/.cache/pre-commit/repo6u_wz1ng/terraform_docs.sh != \/\U\s\e\r\s\/\c\h\r\i\s\.\p\r\z\y\b\y\c\i\e\n\/\.\c\a\c\h\e\/\p\r\e\-\c\o\m\m\i\t\/\r\e\p\o\6\u\_\w\z\1\n\g\/\t\e\r\r\a\f\o\r\m\_\d\o\c\s\.\s\h ]]
+ main . --args=table
+ declare argv
++ getopt -o a: --long args: -- . --args=table
++ _getopt_version_check
++ [[ -z 3.2.57(1)-release ]]
++ [[ 3.2.57(1)-release < 2.05b ]]
++ return 0
++ _getopt_main -o a: --long args: -- . --args=table
++ declare parsed status
++ declare short long name flags
++ declare have_short=false
++ [[ -n '' ]]
++ [[ -o == [^-]* ]]
+++ _getopt_parse getopt ahl:n:o:qQs:TuV alternative,help,longoptions:,name:,options:,quiet,quiet-output,shell:,test,version p -o a: --long args: -- . --args=table
+++ declare name=getopt short=ahl:n:o:qQs:TuV long=alternative,help,longoptions:,name:,options:,quiet,quiet-output,shell:,test,version flags=p
+++ shift 4
+++ declare -a longarr
+++ _getopt_split longarr alternative,help,longoptions:,name:,options:,quiet,quiet-output,shell:,test,version
+++ declare IFS=,
+++ eval 'longarr=( $2 )'
++++ longarr=($2)
+++ longarr=("${longarr[@]/#/--}")
+++ longarr=("${longarr[@]%:}")
+++ longarr=("${longarr[@]%:}")
+++ declare -a opts params
+++ declare o alt_recycled=false error=0
+++ [[ 7 -gt 0 ]]
+++ case $1 in
+++ [[ p == *a* ]]
+++ o=-o
+++ [[ ahl:n:o:qQs:TuV == *\o::* ]]
+++ [[ ahl:n:o:qQs:TuV == *\o:* ]]
+++ [[ 2 -gt 2 ]]
+++ [[ 7 -ge 2 ]]
+++ shift
+++ opts=("${opts[@]}" "$o" "$1")
+++ shift
+++ [[ 5 -gt 0 ]]
+++ case $1 in
+++ o=--long
++++ _getopt_resolve_abbrev --long --alternative --help --longoptions --name --options --quiet --quiet-output --shell --test --version
++++ declare a q=--long
++++ declare -a matches
++++ shift
++++ for a in '"$@"'
++++ [[ --long == \-\-\a\l\t\e\r\n\a\t\i\v\e ]]
++++ [[ p == *a* ]]
++++ [[ --alternative == \-\-\l\o\n\g* ]]
++++ [[ p == *a* ]]
++++ for a in '"$@"'
++++ [[ --long == \-\-\h\e\l\p ]]
++++ [[ p == *a* ]]
++++ [[ --help == \-\-\l\o\n\g* ]]
++++ [[ p == *a* ]]
++++ for a in '"$@"'
++++ [[ --long == \-\-\l\o\n\g\o\p\t\i\o\n\s ]]
++++ [[ p == *a* ]]
++++ [[ --longoptions == \-\-\l\o\n\g* ]]
++++ matches=("${matches[@]}" "$a")
++++ for a in '"$@"'
++++ [[ --long == \-\-\n\a\m\e ]]
++++ [[ p == *a* ]]
++++ [[ --name == \-\-\l\o\n\g* ]]
++++ [[ p == *a* ]]
++++ for a in '"$@"'
++++ [[ --long == \-\-\o\p\t\i\o\n\s ]]
++++ [[ p == *a* ]]
++++ [[ --options == \-\-\l\o\n\g* ]]
++++ [[ p == *a* ]]
++++ for a in '"$@"'
++++ [[ --long == \-\-\q\u\i\e\t ]]
++++ [[ p == *a* ]]
++++ [[ --quiet == \-\-\l\o\n\g* ]]
++++ [[ p == *a* ]]
++++ for a in '"$@"'
++++ [[ --long == \-\-\q\u\i\e\t\-\o\u\t\p\u\t ]]
++++ [[ p == *a* ]]
++++ [[ --quiet-output == \-\-\l\o\n\g* ]]
++++ [[ p == *a* ]]
++++ for a in '"$@"'
++++ [[ --long == \-\-\s\h\e\l\l ]]
++++ [[ p == *a* ]]
++++ [[ --shell == \-\-\l\o\n\g* ]]
++++ [[ p == *a* ]]
++++ for a in '"$@"'
++++ [[ --long == \-\-\t\e\s\t ]]
++++ [[ p == *a* ]]
++++ [[ --test == \-\-\l\o\n\g* ]]
++++ [[ p == *a* ]]
++++ for a in '"$@"'
++++ [[ --long == \-\-\v\e\r\s\i\o\n ]]
++++ [[ p == *a* ]]
++++ [[ --version == \-\-\l\o\n\g* ]]
++++ [[ p == *a* ]]
++++ case ${#matches[@]} in
++++ printf %s --longoptions
++++ return 0
+++ o=--longoptions
+++ [[ ,alternative,help,longoptions:,name:,options:,quiet,quiet-output,shell:,test,version, == *,\l\o\n\g\o\p\t\i\o\n\s,* ]]
+++ [[ ,alternative,help,longoptions:,name:,options:,quiet,quiet-output,shell:,test,version, == *,\l\o\n\g\o\p\t\i\o\n\s\:\:,* ]]
+++ [[ ,alternative,help,longoptions:,name:,options:,quiet,quiet-output,shell:,test,version, == *,\l\o\n\g\o\p\t\i\o\n\s\:,* ]]
+++ [[ 5 -ge 2 ]]
+++ shift
+++ opts=("${opts[@]}" "$o" "$1")
+++ alt_recycled=false
+++ shift
+++ [[ 3 -gt 0 ]]
+++ case $1 in
+++ params=("${params[@]}" "${@:2}")
+++ break
+++ [[ p == *Q* ]]
+++ echo -n ' '
+++ [[ p == *[cu]* ]]
+++ [[ p == *t* ]]
+++ _getopt_quote -o a: --longoptions args: -- . --args=table
+++ declare s space 'q='\'''
+++ for s in '"$@"'
+++ printf ''\''%s'\''' -o
+++ space=' '
+++ for s in '"$@"'
+++ printf ' '\''%s'\''' a:
+++ space=' '
+++ for s in '"$@"'
+++ printf ' '\''%s'\''' --longoptions
+++ space=' '
+++ for s in '"$@"'
+++ printf ' '\''%s'\''' args:
+++ space=' '
+++ for s in '"$@"'
+++ printf ' '\''%s'\''' --
+++ space=' '
+++ for s in '"$@"'
+++ printf ' '\''%s'\''' .
+++ space=' '
+++ for s in '"$@"'
+++ printf ' '\''%s'\''' --args=table
+++ space=' '
+++ echo
+++ return 0
++ parsed=' '\''-o'\'' '\''a:'\'' '\''--longoptions'\'' '\''args:'\'' '\''--'\'' '\''.'\'' '\''--args=table'\'''
++ status=0
++ [[ 0 != 0 ]]
++ eval 'set --  '\''-o'\'' '\''a:'\'' '\''--longoptions'\'' '\''args:'\'' '\''--'\'' '\''.'\'' '\''--args=table'\'''
+++ set -- -o a: --longoptions args: -- . --args=table
++ [[ 7 -gt 0 ]]
++ case $1 in
++ short=a:
++ have_short=true
++ shift
++ shift
++ [[ 5 -gt 0 ]]
++ case $1 in
++ long=args:
++ shift
++ shift
++ [[ 3 -gt 0 ]]
++ case $1 in
++ shift
++ break
++ true
++ [[ a: == -* ]]
++ [[ a: == +* ]]
++ flags=
++ _getopt_parse getopt a: args: '' . --args=table
++ declare name=getopt short=a: long=args: flags=
++ shift 4
++ declare -a longarr
++ _getopt_split longarr args:
++ declare IFS=,
++ eval 'longarr=( $2 )'
+++ longarr=($2)
++ longarr=("${longarr[@]/#/--}")
++ longarr=("${longarr[@]%:}")
++ longarr=("${longarr[@]%:}")
++ declare -a opts params
++ declare o alt_recycled=false error=0
++ [[ 2 -gt 0 ]]
++ case $1 in
++ [[ '' == *i* ]]
++ [[ '' == *p* ]]
++ params=("${params[@]}" "$1")
++ shift
++ [[ 1 -gt 0 ]]
++ case $1 in
++ o=--args
+++ _getopt_resolve_abbrev --args --args
+++ declare a q=--args
+++ declare -a matches
+++ shift
+++ for a in '"$@"'
+++ [[ --args == \-\-\a\r\g\s ]]
+++ matches=("$a")
+++ break
+++ case ${#matches[@]} in
+++ printf %s --args
+++ return 0
++ o=--args
++ [[ ,args:, == *,\a\r\g\s::,* ]]
++ [[ ,args:, == *,\a\r\g\s:,* ]]
++ opts=("${opts[@]}" "$o" "${1#*=}")
++ alt_recycled=false
++ shift
++ [[ 0 -gt 0 ]]
++ [[ '' == *Q* ]]
++ echo -n ' '
++ [[ '' == *[cu]* ]]
++ [[ '' == *t* ]]
++ _getopt_quote --args table -- .
++ declare s space 'q='\'''
++ for s in '"$@"'
++ printf ''\''%s'\''' --args
++ space=' '
++ for s in '"$@"'
++ printf ' '\''%s'\''' table
++ space=' '
++ for s in '"$@"'
++ printf ' '\''%s'\''' --
++ space=' '
++ for s in '"$@"'
++ printf ' '\''%s'\''' .
++ space=' '
++ echo
++ return 0
++ declare status=0
++ unset -f _getopt_main _getopt_err _getopt_parse _getopt_quote _getopt_quote_csh _getopt_resolve_abbrev _getopt_split _getopt_help _getopt_version_check
++ return 0
+ argv=' '\''--args'\'' '\''table'\'' '\''--'\'' '\''.'\'''
+ eval 'set --  '\''--args'\'' '\''table'\'' '\''--'\'' '\''.'\'''
++ set -- --args table -- .
+ declare args
+ declare files
+ for argv in '"$@"'
+ case $argv in
+ shift
+ args=table
+ shift
+ for argv in '"$@"'
+ case $argv in
+ for argv in '"$@"'
+ case $argv in
+ shift
+ files=.
+ break
+ local hack_terraform_docs
++ terraform version
++ head -1
++ grep -c 0.12
+ hack_terraform_docs=1
++ command -v terraform-docs
+ [[ ! -n /usr/local/bin/terraform-docs ]]
+ local is_old_terraform_docs
++ terraform-docs version
++ grep -o 'v0.[1-7]'
++ tail -1
+ is_old_terraform_docs=v0.1
+ [[ -z v0.1 ]]
+ [[ 1 == \1 ]]
++ command -v awk
+ [[ ! -n /usr/local/bin/awk ]]
++ mktemp /var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-XXXXXXXXXX
+ tmp_file_awk=/var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-ptPb5onKgZ
+ terraform_docs_awk /var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-ptPb5onKgZ
+ readonly output_file=/var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-ptPb5onKgZ
+ output_file=/var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-ptPb5onKgZ
+ cat
+ terraform_docs /var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-ptPb5onKgZ table .
+ readonly terraform_docs_awk_file=/var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-ptPb5onKgZ
+ terraform_docs_awk_file=/var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-ptPb5onKgZ
+ readonly args=table
+ args=table
+ readonly files=.
+ files=.
+ declare -a paths
+ declare -a tfvars_files
+ index=0
+ for file_with_path in '$files'
+ file_with_path=.
++ dirname .
+ paths[index]=.
+ [[ . == *\.\t\f\v\a\r\s ]]
+ (( index += 1 ))
++ mktemp
+ readonly tmp_file=/var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T/tmp.xYdZ1gOp
+ tmp_file=/var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T/tmp.xYdZ1gOp
+ readonly text_file=README.md
+ text_file=README.md
++ echo .
++ tr ' ' '\n'
++ sort -u
+ for path_uniq in '$(echo "${paths[*]}" | tr '\'' '\'' '\''\n'\'' | sort -u)'
+ path_uniq=.
+ pushd .
+ [[ ! -f README.md ]]
+ [[ /var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-ptPb5onKgZ == \0 ]]
++ mktemp /var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-XXXXXXXXXX
+ tmp_file_docs=/var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-Zn87UoxBoz
+ mv /var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-Zn87UoxBoz /var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-Zn87UoxBoz.tf
+ tmp_file_docs_tf=/var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-Zn87UoxBoz.tf
+ awk -f /var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-ptPb5onKgZ ./main.tf ./output.tf ./variables.tf
+ terraform-docs md table /var/folders/d7/jgsn0b011l5fk4vnmhvbt0tr0000gp/T//terraform-docs-Zn87UoxBoz.tf
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x13fdbe6]

goroutine 1 [running]:
github.com/terraform-docs/terraform-docs/internal/cli.(*cfgreader).exist(0xc00014c1c0, 0xc00014c1c0, 0x2, 0xc00006a310)
	/private/tmp/terraform-docs-20200921-68242-fhrnj8/terraform-docs-0.10.0/internal/cli/reader.go:22 +0xc6
github.com/terraform-docs/terraform-docs/internal/cli.PreRunEFunc.func1(0xc0001b6000, 0xc0001b4360, 0x1, 0x1, 0x0, 0x0)
	/private/tmp/terraform-docs-20200921-68242-fhrnj8/terraform-docs-0.10.0/internal/cli/run.go:46 +0x1b9
github.com/spf13/cobra.(*Command).execute(0xc0001b6000, 0xc0001b4340, 0x1, 0x1, 0xc0001b6000, 0xc0001b4340)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:831 +0x530
github.com/spf13/cobra.(*Command).ExecuteC(0xc00013eb00, 0x1044337, 0x17630e0, 0xc000010080)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/terraform-docs/terraform-docs/cmd.Execute(0x1434440, 0xc0000160b8)
	/private/tmp/terraform-docs-20200921-68242-fhrnj8/terraform-docs-0.10.0/cmd/root.go:24 +0x2b
main.main()
	/private/tmp/terraform-docs-20200921-68242-fhrnj8/terraform-docs-0.10.0/main.go:10 +0x25
```
</details>

--- a/internal/cli/reader.go
+++ b/internal/cli/reader.go
@@ -19,7 +19,7 @@ func (c *cfgreader) exist() (bool, error) {
 	if c.file == "" {
 		return false, fmt.Errorf("config file name is missing")
 	}
-	if info, err := os.Stat(c.file); os.IsNotExist(err) || info.IsDir() {
+	if info, err := os.Stat(c.file); os.IsNotExist(err) || info == nil || info.IsDir() {
 		return false, err
 	}
 	return true, nil

--- a/internal/cli/reader_test.go
+++ b/internal/cli/reader_test.go
@@ -36,6 +36,13 @@ func TestExists(t *testing.T) {
 			wantErr:  true,
 			errMsg:   "config file name is missing",
 		},
+		{
+			name:     "main argument is a file",
+			config:   cfgreader{file: "testdata/sample-config.yaml/some-config.yaml"},
+			expected: false,
+			wantErr:  true,
+			errMsg:   "stat testdata/sample-config.yaml/some-config.yaml: not a directory",
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {

